packages/compass-aggregations/src/stores/store.ts

```diff
@@ -258,6 +259,28 @@ export function activateAggregationsPlugin(
     maxTimeMSChanged(preferences.getPreferences().maxTimeMS || null)
   );
 
+  addCleanup(
+    workspaces.registerTabCloseHandler?.(async function (nextTab) {
+      if (!store.getState().isModified) {
+        return 'allow';
+      }
+
+      if (nextTab) {
+        return 'open-in-new-tab';
+      }
+
+      const confirmed = await showConfirmation({
+        title: 'Your aggregation contains unsaved changes',
+        description:
+          'You will loose all unsaved changes to the pipeline you are currently building',
+        buttonText: 'Close tab',
+        variant: 'danger',
+      });
+
+      return confirmed ? 'user-allow' : 'deny';
```

Comment:
That's the one that is special to aggergations, that asks for user confirmation before closing the tab (as signalled by `nextTab` being `null`)